documentation/docs/developer_guide/coding_standards.md

```diff
@@ -1 +1,179 @@
+
+### 6. MODULE names
+
+Everything should be contained within modules.
+Modules should be named of the form `*_mod_cbl`. Following from the above example, the file `egName_cbl.F90` should contain the module declaration `MODULE egName_mod_cbl`
+
+### 7. SUBROUTINE names
+
+Subroutines should be of the form "\*_cbl()". Following from the above file `egName_cbl.F90` `SUBROUTINE egName_cbl()`. This is not only helpful for consistency but necessary within JULES/UM applications where it distinguishes the CABLE version of an analogous JULES subroutine. 
+Markdown needs to escape * characters.
+
+### 8. Miscellaneous
+
+ * The last character of a line **must** not be placed beyond column 80. 
+ * `SAVE` attributes are not permitted.
+ * Intrinsic Fortran subroutines, functions, etc should be capitalized
+ * Subroutines start on the margin. This includes MODULE, CONTAINS, SUBROUTINE, USE statements, and declarations. Indentation of two spaces occurs in loops/layers. This is vital in readily identifying the extent of loops/layers. i.e. Under which condition(s) a section of code is relevantin an IF layer, or at what index is being traversed in a DO loop. 
+
+ * Specify `IMPLICIT NONE` in all modules to prevent type mismatch errors 
+ * Dummy arguments must be declared with the `INTENT()` attribute
+ * Declare in the `MODULE` preamble which elements are `PUBLIC`
+ * Import specific components using `ONLY` with `USE` statements to avoid inadvertent inheritance and
+   increase transparency of underlying data flow 
+ * Label each side of a loop, especially if it extends
+   more than a page.
+ * All `IF` constructs must be of the form:
+```fortran
+IF() THEN 
+  Even if they only contain a simple, single line statement following the IF. 
+END IF
+```
+ * In summary, be pedantic, be clear
+
+## Final Note about JAC
+
+Scientific models have naturally evolved over time to include increasing complexity. Coupled with the demand for higher resolution, larger ensembles etc, this has required ever increasing computational performance.
+For several decades, this icreasing performance demand has been met by technological advances in hardware.   
+This era has come to an end.
+The Next Generation Modelling System (NGMS) is a major program at the UKMO to address this issue.
+The NGMS is an initiative to optimize the software of their entire suite of models.
+The specific coding standards required by the NGMS have not yet been strictly defined. 
+
+In general, models (including CABLE) have evolved over decades, involving dozens of developers. A key feature emerging from NGMS is the tight reign on memory. As such, it will not be permitted to `USE` data through `MODULE`s. Data will be be passed through argument lists. There are further rules emerging in this respect which relax the rules slightly to allow time independent scalars to be `USE`d. Nevertheless, it is straightforward enough to pass these as well. 
```

Comment:
Do we need this?

The rule to pass arguments instead of `USE` should be stated as a rule and not lost in a full paragraph.

Reply:
listed as a rule. 

see - dc5b6d7e9